File: src/main/java/cn/xgp/xgplottery/Utils/VersionAdapterUtils.java
package cn.xgp.xgplottery.Utils;

import org.bukkit.entity.Player;
import org.bukkit.event.player.PlayerInteractEvent;
import org.bukkit.inventory.EquipmentSlot;
import org.bukkit.inventory.ItemStack;

//for legacy version
public class VersionAdapterUtils {

    public static boolean ifMainHand(PlayerInteractEvent e){
        return EquipmentSlot.HAND.equals(e.getHand());
    }
    public static void setItemInMainHand(Player player,ItemStack itemStack){
        player.getInventory().setItemInMainHand(itemStack);
    }

    public static ItemStack getItemInMainHand(Player player){
        return player.getInventory().getItemInMainHand();
    }
    public static int getPlayerEmptySlot(Player player){
        ItemStack[] items = player.getInventory().getContents();
        int emptySlots = 0;
        for (ItemStack i : items) {
            if (i == null) {
                emptySlots++;
            }
        }
        ItemStack[] equipment = player.getEquipment().getArmorContents();
        for (ItemStack i : equipment) {
            if (i == null) {
                emptySlots--;
            }
        }
        if(player.getInventory().getItemInOffHand().getItemMeta() == null)
            emptySlots--;
        return emptySlots;

    }
}

File: src/main/kotlin/cn/xgpjun/xgplottery2/utils/VersionAdapterUtils.kt
package cn.xgpjun.xgplottery2.utils

import cn.xgpjun.xgplottery2.manager.NMSManager
import org.bukkit.entity.Player
import org.bukkit.event.player.PlayerInteractEvent
import org.bukkit.inventory.EquipmentSlot
import org.bukkit.inventory.ItemStack

class VersionAdapterUtils {
    companion object{
        fun getPlayerEmptySlot(player: Player): Int {
            val items = player.inventory.contents
            var emptySlots = 0
            for (i in items) {
                if (i == null) {
                    emptySlots++
                }
            }
            val equipment = player.equipment!!.armorContents
            for (i in equipment) {
                if (i == null) {
                    emptySlots--
                }
            }
            return if (NMSManager.versionToInt < 9) {
                emptySlots
            } else {
                if (player.inventory.itemInOffHand.itemMeta == null) emptySlots--
                emptySlots
            }
        }
    }
}

fun Player.setItemInMainHand(itemStack: ItemStack?) {
    if (NMSManager.versionToInt < 9) {
        this.setItemInHand(itemStack)
    } else {
        this.inventory.setItemInMainHand(itemStack)
    }
}

fun Player.getItemInMainHand(): ItemStack {
    return if (NMSManager.versionToInt < 9) this.itemInHand else this.inventory.itemInMainHand
}

fun PlayerInteractEvent.isMainHand():Boolean{
    return if (NMSManager.versionToInt < 9) true else EquipmentSlot.HAND == this.hand
}